src/mcedit2/panels/worldmeta.py

```diff
@@ -0,0 +1,255 @@
+"""
+    player
+"""
+from __future__ import absolute_import
+import logging
+from contextlib import contextmanager
+
+from PySide import QtGui
+from PySide.QtCore import Qt
+
+from mcedit2.util.resources import resourcePath
+from mcedit2.util.load_ui import load_ui
+from mcedit2.util.screen import centerWidgetInScreen
+from mcedit2.widgets.nbttree.nbteditor import NBTDataChangeCommand
+
+log = logging.getLogger(__name__)
+
+
+class WorldMetaPanel(QtGui.QWidget):
+    GENERATOR_TYPES = ["default", "flat", "largeBiomes", "amplified", "customized", 'debug_all_block_states']
+    GAMEMODES = ["Survival", "Creative", "Adventure", "Spectator"]
+    DIFFICULTIES = ["Peaceful", "Easy", "Normal", "Hard"]
+
+    editsDisabled = False
+
+    def __init__(self, editorSession):
+        super(WorldMetaPanel, self).__init__(QtGui.qApp.mainWindow, f=Qt.Tool)
+        self.editorSession = editorSession
+        self.worldMeta = editorSession.worldEditor.adapter.metadata
+        callIcon = QtGui.QIcon(resourcePath("mcedit2/assets/mcedit2/icons/edit_metadata.png"))
+        callButton = QtGui.QAction(callIcon, "Edit World Metadata", self)
+        callButton.setCheckable(True)
+        callButton.triggered.connect(self.toggleView)
+        self._toggleViewAction = callButton
+
+        load_ui("panels/worldmeta.ui", baseinstance=self)
+
+        self.worldNBTEditor.editorSession = self.editorSession
+        self.editorSession.revisionChanged.connect(self.revisionDidChange)
+
+        self.updateNBTTree()
+        self.setupPanel()
+        self.updatePanel()
+
+        centerWidgetInScreen(self)
+
+    def updatePanel(self):
+        self.defaultGamemodeCombo.setCurrentIndex(self.worldMeta.GameType)
+        self.worldDifficultyCombo.setCurrentIndex(self.worldMeta.Difficulty)
+        self.generationTypeCombo.setCurrentIndex(self.GENERATOR_TYPES.index(self.worldMeta.generatorName))
+
+        self.worldNameLineEdit.setText(self.worldMeta.LevelName)
+        self.worldNameLineEdit.setText(self.worldMeta.LevelName)
+        self.generatorSeedLineEdit.setText(unicode(str(self.worldMeta.RandomSeed), encoding='utf-8'))
```

Comment:
This unicode decode isn't needed.

I double checked - str(int), str(long) and str(float) don't honor the current locale. They always return ASCII-encoded strings so they can be passed to `setText` as-is. The only time you get a locale-aware conversion from numbers to strings is if you `import locale` and use its conversion funcs.

Qt probably has its own locale conversions somewhere, too.